### PR TITLE
Add doc for impl From for Addr

### PR DESCRIPTION
**Pull request by phungleson**
As part of issue #51430 (cc @skade).

The impl is very simple, let me know if we need to go into any details.

Additionally, I added `#[inline]` for the conversion method, let me know if it is un-necessary or might break something.

### CODE REVIEW

File: src/libstd/net/addr.rs [hunk 1]
```diff
@@ -554,20 +554,44 @@ impl FromInner<c::sockaddr_in6> for SocketAddrV6 {
 
 #[stable(feature = "ip_from_ip", since = "1.16.0")]
 impl From<SocketAddrV4> for SocketAddr {
+    /// Converts a [`SocketAddrV4`] into a [`V4 variant of SocketAddr`].
+    ///
+    /// [`SocketAddrV4`]: struct.SocketAddrV4.html
+    /// [`V4 variant of SocketAddr`]: enum.SocketAddr.html#variant.V4
```
Comment (phungleson) [inline src/libstd/net/addr.rs]: Sorry for my noob question as I haven't touched rust for quite a while.

Is this syntax valid and necessary for the docs?

Comment (dtolnay) [reply src/libstd/net/addr.rs]: Based on running `cargo doc` on the following, yes I believe it is valid as written.

```rust
pub enum SocketAddr {
    V4,
}

/// Converts a `SocketAddrV4` into a [`V4 variant of SocketAddr`].
///
/// [`V4 variant of SocketAddr`]: enum.SocketAddr.html#variant.V4
pub fn from() {}
```

A more modern way to write it as of https://github.com/rust-lang/rust/issues/43466 would be:

```rust
/// [`V4 variant of SocketAddr`]: SocketAddr::V4
```

**Comment by Mark-Simulacrum** [comment]
The inline attribute should be removed, or at least discussed in a separate PR. Those should only be added given real-world benchmarks demonstrating need.

**Comment by pietroalbini** [comment]
Ping from triage @skade! This PR needs your review.

**Comment by phungleson** [comment]
Hey, I am still traveling, I will fix this when I am back, soon.

**Pull request merged**